src/components/rules-creation-page/steps/IncludeExcludeSelectionStep.tsx

```diff
@@ -0,0 +1,236 @@
+            const models = _.keys(metadata).map((type: string) => {
+                return d2ModelFactory(d2, type);
+            });
+
+            const parseModels = (models: typeof D2Model[]) =>
+                models
+                    .map((model: typeof D2Model) => model.getD2Model(d2))
+                    .map((model: ModelDefinition) => ({
+                        name: model.displayName,
+                        id: model.name,
+                    }));
+
+            setModels(models);
+            setModelSelectItems(parseModels(models));
+        });
+
+        console.log(syncRule.metadataExcludeIncludeRules);
+    }, [d2, syncRule]);
+
+    const changeUseDefaultIncludeExclude = (useDefault: boolean) => {
+        onChange(
+            useDefault
+                ? syncRule.markToUseDefaultIncludeExclude()
+                : syncRule.markToNotUseDefaultIncludeExclude(models)
+        );
+    };
+
+    const changeModelName = (event: any) => {
+        setSelectedType(event.target.value);
+    };
+
+    const changeInclude = (includeRules: any) => {
+        const type: string = selectedType ?? "";
+
+        const oldIncludeRules = getIncludeRules();
+        const oldExcludeRules = getExcludeRules();
+
+        const ruleIndexesToExclude = _.difference(oldIncludeRules, includeRules).map(rule =>
+            oldIncludeRules.indexOf(rule)
+        );
+        const ruleIndexesToInclude = _.difference(includeRules, oldIncludeRules).map(rule =>
+            oldExcludeRules.indexOf(rule)
+        );
+
+        if (ruleIndexesToInclude.length > 0) {
+            onChange(syncRule.moveRuleFromExcludeToInclude(type, ruleIndexesToInclude));
+        } else if (ruleIndexesToExclude.length > 0) {
+            onChange(syncRule.moveRuleFromIncludeToExclude(type, ruleIndexesToExclude));
+        }
+    };
+
+    const getAllRules = () => {
+        const allRules =
+            selectedType && syncRule.metadataExcludeIncludeRules
+                ? [
+                      ...syncRule.metadataExcludeIncludeRules[selectedType].excludeRules,
+                      ...syncRule.metadataExcludeIncludeRules[selectedType].includeRules,
+                  ]
+                : [];
+
+        return allRules;
+    };
+
+    const getIncludeRules = () => {
+        return selectedType && syncRule.metadataExcludeIncludeRules
+            ? syncRule.metadataExcludeIncludeRules[selectedType].includeRules
+            : [];
+    };
+
+    const getExcludeRules = () => {
+        return selectedType && syncRule.metadataExcludeIncludeRules
+            ? syncRule.metadataExcludeIncludeRules[selectedType].excludeRules
+            : [];
+    };
+
+    const getRuleOptions = () => {
+        const allRules = getAllRules();
+
+        return allRules.map(rule => ({
+            value: rule,
+            text: includeExcludeRulesFriendlyNames[rule]
+                ? includeExcludeRulesFriendlyNames[rule]
+                : rule,
```

Comment:
[minor] ``includeExcludeRulesFriendlyNames[rule] ?? rule``

Reply:
ok I'll do it